### PR TITLE
XWIKI-20546: Log CSS exception details

### PR DESCRIPTION
**Pull request by amottier**
# Jira URL

https://jira.xwiki.org/browse/XWIKI-20546

# Changes

## Description

* In XWikiWikiModel, if parsing CSS exception occurs, includes the exception in SLF4J call to get the full stack trace in the logs. This should help investigation of [XWIKI-20546](https://jira.xwiki.org/browse/XWIKI-20546) issue.

## Clarifications

* Root cause of the issue might be related to thread safety issue with `CSSOMParser`.

# Executed Tests

`mvn -f xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-xwiki/pom.xml clean install` with Java 17 in Linux environment.

# Expected merging strategy

* Prefers squash: Yes
* Backport on branches: No

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-xwiki/src/main/java/org/xwiki/rendering/internal/wiki/XWikiWikiModel.java [hunk 1]
```diff
@@ -312,7 +312,7 @@ private String getImageDimension(String dimension, Map<String, String> imagePara
                 value = sd.getPropertyValue(dimension);
             } catch (Exception e) {
                 // Ignore the style parameter but log a warning to let the user know.
-                this.logger.warn("Failed to parse CSS style [{}]", style);
+                this.logger.warn("Failed to parse CSS style [{}]", style, e);
```
Comment (tmortagne) [inline xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-xwiki/src/main/java/org/xwiki/rendering/internal/wiki/XWikiWikiModel.java:315]: We tend to only log a `org.apache.commons.lang3.exception.ExceptionUtils.getRootCauseMessage(e)` in the case of a warning (a printed exception is pretty huge in the log, so we do that only for errors).

Comment (amottier) [reply xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-xwiki/src/main/java/org/xwiki/rendering/internal/wiki/XWikiWikiModel.java:315]: Thanks @tmortagne for clarifying this coding rule.

Actually what I'm might do is deploying this custom version only in our development environment, reproduce the issue described in [XWIKI-20546](https://jira.xwiki.org/browse/XWIKI-20546) and add the log in the Jira ticket comments.

I'm closing this pull request. Thanks for taking the time to look at it.

Comment (tmortagne) [reply xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-xwiki/src/main/java/org/xwiki/rendering/internal/wiki/XWikiWikiModel.java:315]: Note that I feel that adding at least the root cause message would be an improvement in any case, it's definitely not great to not have any kind of clue of what was wrong.

Comment (amottier) [reply xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-xwiki/src/main/java/org/xwiki/rendering/internal/wiki/XWikiWikiModel.java]: I'm reopening the pull request this time logging only the root cause.

Also I took the time to read https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/ and properly configure my IDE.

**Comment by tmortagne** [comment]
Thanks a lot for your time @amottier !

That being said, my understanding is that it's not really a fix for https://jira.xwiki.org/browse/XWIKI-20546 and that we need to fix what cause this failure in the first place, right ?

**Comment by amottier** [comment]
Yes you are totally right @tmortagne, main purpose of this pull request is to get appropriate level of information to be able to further analyze the issue.

**Pull request merged**